qhub/schema.py

```diff
@@ -364,6 +364,27 @@ class Main(Base):
     environments: typing.Dict[str, CondaEnvironment]
     clearml: typing.Optional[ClearML]
 
+    @validator("project_name")
+    def project_name_convention(cls, value: typing.Any):
+        convention = """
+        In order to successfully deploy QHub, there are some project naming conventions which need
+        to be followed. First, ensure your name is compatible with the specific one for
+        your chosen Cloud provider. In addition, the QHub project name should also obey the following
+        format requirements:
+        - letters from A to Z (upper and lower case) and numbers;
+        - Special characters are not allowed, such as "!@#$%^&*()";
+        - Maximum accepted length of the name string is 16 characters.
+        - If using AWS: names should not start with the string "aws";
+        """
+        if len(value) > 16:
+            raise ValueError(
+                "Maximum accepted length of the project name string is 16 characters."
+            )
+        elif re.findall(r"^(?!aws)[A-Za-z0-9][^/|.~!?@#$%^=&*\\()_]*[A-Za-z0-9]$", value):
```

Comment:
@tarundmsharma so right now this regex is not considering all dashes in the project name, which is not something we would like to have for all providers... only for Azure. So we need to add a conditional that checks the target provider and use the correct regex based on it.